syntax = "proto3";

package storage;

message UploadVideoRequest {
    string id = 1;
    uint32 version = 2;
    bytes data = 3;
}

message UploadVideoResponse {
    bool success = 1;
}

message DownloadVideoRequest {
    string id = 1;
    uint32 version = 2;
}

message DownloadVideoResponse {
    bool success = 1;
    bytes data = 2;
}

message DeleteVideoRequest {
    string id = 1;
    uint32 version = 2;
}

message DeleteVideoResponse {
    bool success = 1;
}
